tests/unit/apps/test_auxiliary.py

```diff
@@ -1447,3 +1443,47 @@ def test_expected_current_channel_auxiliary(mock_os_release, model, channel, ori
     )
     # expected_current_channel is indifferent if the charm needs crossgrade
     assert ceph_osd.expected_current_channel(target) == "octopus/stable"
+
+
+def test_auxiliary_wrong_channel(model):
+    """Test when an auxiliary charm is with a channel that doesn't match the workload version."""
+    target = OpenStackRelease("victoria")
+    charm = "ceph-mon"
+    machines = {"0": MagicMock(spec_set=Machine)}
+    app = CephMon(
+        name=charm,
+        can_upgrade_to="",
+        charm=charm,
+        channel="quincy/stable",
+        config={"source": {"value": "distro"}},
+        machines=machines,
+        model=model,
+        origin="ch",
+        series="focal",
+        subordinate_to=[],
+        units={
+            f"{charm}/0": Unit(
+                name=f"{charm}/0",
+                workload_version="15.2.0",
+                machine=machines["0"],
+            )
+        },
+        workload_version="15.2.0",
+    )
+
+    # plan will warn that channel will change from quincy to octopus to match the workload version,
+    # but this can be a downgrade.
+    exp_plan = dedent_plan(
+        """\
+    Upgrade plan for 'ceph-mon' to 'victoria'
+        Upgrade software packages of 'ceph-mon' from the current APT repositories
+            Ψ Upgrade software packages on unit 'ceph-mon/0'
+        Ensure that the 'require-osd-release' option matches the 'ceph-osd' version
+        WARNING: Changing 'ceph-mon' channel from quincy/stable to octopus/stable. This may be a charm downgrade, which is generally not supported.
```

Comment:
not=blocking: You can use `\` as you did in first line instead of adding `# noqa: E501 line too long`